website/integrations: add Wazuh (#7706)
goauthentik/authentik: website/integrations/services/wazuh/index.md

![](./provider5.png)

### Step 3 - property mapping

Time to create a Property Mapping - this is a custom function that takes group/user data from authentik and provides it to wazuh in a structured way.

We will map a group membership - `wazuh-admins` - as a backend role for RBAC in wazuh using Property Mapping - `Customization` -> `Property Mappings`

`Name: wazuh property mapping`

`SAML Attribute Role: Roles`

![](./property-mapping.png)
![](./property-mapping2.png)

```python
if ak_is_group_member(request.user, name="wazuh-admins"):
  yield "wazuh-admin"
```

Make sure to adjust the provider to include the newly created property mapping in the `Advanced protocol settings`.
![](./property-mapping3.png)

### Step 4 - saml application

Now create an application to use the newly created provider. `Applications` -> `Applications` -> `Create`

`Name: wazuh-saml`
`Slug: wazuh-saml`
`Provider: SAML`
`Policy Engine: any`

![](./application.png)

You can change the UI / upload a logo so that in the applications overview you have a nice layout and can easily identify the new wazuh app.
![](./applications_overview.png)

### Step 5 - metadata + wazuh opensearch-security configuration

> Now download the metadata file `saml_authentik_meta.xml` from the `Applications` -> `Provider` -> `Related Objects` -> `Download`

Author: this is in authentik - you download the xml file with the tenant/provider configuration from authentik - in theory there should also be a screenshot showing where exactly to click if I remember correctly but I will make sure it is there once more

-   make sure to adjust the ownership and access rights via the following commands:

```bash
sudo chown wazuh-indexer:wazuh-indexer /etc/wazuh-indexer/opensearch-security/idp-metadata.xml

sudo chmod 640 /etc/wazuh-indexer/opensearch-security/idp-metadata.xml
```

```yml
authc:
  basic_internal_auth_domain:
    description: "Authenticate via HTTP Basic + SAML against internal users database"
    http_enabled: true
    transport_enabled: true b
    order: 0
    http_authenticator:
      type: basic
      challenge: false
    authentication_backend:
      type: intern
  saml_auth_domain:
    http_enabled: true
    transport_enabled: false
    order: 1
    http_authenticator:
      type: saml
      challenge: true
      config:
        idp:
          metadata_file: "/etc/wazuh-indexer/opensearch-security/idp-metadata.xml"
          entity_id: "wazuh-saml"
        sp:
          entity_id: "wazuh-saml"
        kibana_url: "https://<YOUR_WAZUH_IP_OR_HOSTNAME>"
        roles_key: Roles
        exchange_key: "MIIGIODCCA+ygAwIBAgITREIpdNn++usuy4Gq5g0W2/SQs..."

>      authentication_backend:

Reviewer: ```suggestion
      authentication_backend:
```

>        type: noop

Reviewer 2: This is not valid YAML, the indentation is wrong

```

### Step 6 - securityadmin.sh

> After you save the `config.yml` you need to restart the wazuh manager and tell it to load the modified file - this can be done automagically using the `securityadmin.sh` wazuh provides with the following command:

Reviewer 3: ```suggestion
After you save the `config.yml` you need to restart the wazuh manager and tell it to load the modified file - this can be done automagically using the `securityadmin.sh` file that wazuh provides with the following command:
```

```bash
export JAVA_HOME=/usr/share/wazuh-indexer/jdk/ && bash /usr/share/wazuh-indexer/plugins/opensearch-security/tools/securityadmin.sh -f /etc/wazuh-indexer/opensearch-security/config.yml -icl -key /etc/wazuh-indexer/certs/admin-key.pem -cert /etc/wazuh-indexer/certs/admin.pem -cacert /etc/wazuh-indexer/certs/root-ca.pem -h localhost -nhnv
```

> If all goes well this shows `Done with success` in the end

Reviewer 3: ```suggestion
If all goes well, `Done with success` displays.
```

### Step 7 - roles_mapping.yml

> In order to map the backend roles from authentik to wazuh specific roles you need to adjust the `/etc/wazuh-indexer/opensearch-security/roles_mapping.yml`

Reviewer 3: ```suggestion
In order to map the backend roles from authentik to wazuh-specific roles you need to modify the `/etc/wazuh-indexer/opensearch-security/roles_mapping.yml` file.
```

Open the file and scroll to the following section:
![Alt text](roles_mapping.png)

under `backend_roles` add `wazuh-admin` and check that reserved is `false` + the rest looks like below:

```yml
all_access:
    reserved: false
    hidden: false
    backend_roles:
        - "wazuh-admin"
        - "admin"
    hosts: []
    users: []
    and_backend_roles: []
    description: "Maps admin to all_access"
```

> save the file and use the securityadmin.sh with the following command to load the adjusted `roles_mapping.yml`:

Reviewer 3: ```suggestion
Save the file and use the `securityadmin.sh` file with the following command to load the adjusted `roles_mapping.yml`:
```

```bash
export JAVA_HOME=/usr/share/wazuh-indexer/jdk/ && bash /usr/share/wazuh-indexer/plugins/opensearch-security/tools/securityadmin.sh -f /etc/wazuh-indexer/opensearch-security/roles_mapping.yml -icl -key /etc/wazuh-indexer/certs/admin-key.pem -cert /etc/wazuh-indexer/certs/admin.pem -cacert /etc/wazuh-indexer/certs/root-ca.pem -h localhost -nhnv
```

> If all goes well this shows `Done with success` in the end

Reviewer 3: ```suggestion
If all goes well, `Done with success` displays.
```

### Step 8 - wazuh.yml

Check `/usr/share/wazuh-dashboard/data/wazuh/config/wazuh.yml` and make sure that `run_as` is set to `false`.
![](wazuhyml.png)

### Step 9 - security role mapping

> open the wazuh dashboard - click on the downward pointing triangle next to the wazuh logo then on `Security` and `Roles mapping`.

Reviewer 3: ```suggestion
Open the wazuh dashboard, click on the downward pointing triangle next to the wazuh logo, and then click on **Security** and **Roles mapping**.
```

![](roles_mapping1.png)

> We will now add a new role mapping - add any name + the respective Roles -> in this case administrator and add a new custom rule at the bottom that matches (`FIND`) the `user_name` to `wazuh-admin`.

Reviewer 3: ```suggestion
We will now add a new role mapping. Give it any descriptive name and then add the respective Roles. In this case `administrator` is the name,  and add a new custom rule at the bottom that matches (`FIND`) the `user_name` to `wazuh-admin`.
```

![](saml-admin.png)

### Step 10 - final step - opensearch_dashboards.yml

> The last step is to adapt the `/etc/wazuh-dashboard/opensearch_dashboards.yml` and add three lines to the bottom of the file:

Reviewer 3: ```suggestion
The last step is to modify `/etc/wazuh-dashboard/opensearch_dashboards.yml` and add three lines to the bottom of the file:
```

Reviewer 3: "adapt" would work as a verb, if you prefer it; we usually say wither "edit" or "modify" when we write about files.

```yml
opensearch_security.auth.type: "saml"
server.xsrf.allowlist:
    [
        "/_opendistro/_security/saml/acs",
        "/_opendistro/_security/saml/logout",

>         "/_opendistro/_security/saml/acs/idpinitiated",

Reviewer 4: Additional comma at the end needs to be removed :)

    ]
opensearch_security.session.keepalive: false
```

![](dashboardsyml.png)

> If all went well you should now only have to restart the wazuh dashboard

Reviewer 3: ```suggestion
If all went well you should now only have to restart the wazuh dashboard.
```

`systemctl restart wazuh-dashboard`

> and then you should be greeted by the authentik login screen when you try to connect to the dashboard

Reviewer 3: ```suggestion
And then you should be greeted by the authentik login screen when you try to connect to the dashboard.
```
